tests/SelfTest/IntrospectiveTests/Clara.tests.cpp

```diff
@@ -51,6 +51,16 @@ TEST_CASE("Clara::Arg supports single-arg parse the way Opt does", "[clara][arg]
     REQUIRE(name == "foo");
 }
 
+TEST_CASE("Clara::Arg does not crash on incomplete input", "[clara][arg][compilation]") {
+    std::string name;
+    auto p = Catch::Clara::Arg(name, "-");
+
+    CHECK(name.empty());
+
+    p.parse( Catch::Clara::Args{ "UnitTest", "-" } );
```

Comment:
This should probably check that the parsing has failed.

Reply:
I added some extra tests and updated the Approved stuff.